packaging/build-container.sh

```diff
@@ -0,0 +1,67 @@
+#! /bin/bash -e
+#
+# Copyright 2019 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+EXEC_PATH="$(dirname "$(realpath "$0")")"
+PROJECT_PATH="$(dirname $EXEC_PATH)"
+
+options=$(getopt --options "" \
+    --longoptions extra-args: \
+    -- "${@}")
+
+
+eval set -- "${options}"
+while :
+do
+    case "${1}" in
+        --extra-args)
+            shift
+            extra_args="${1}"
+            ;;
+        --)
+            shift
+            break
+            ;;
+    esac
+    shift
+done
+
+rebuild_container() {
+    local container_name
+    local extra_args
+
+    extra_args="${1}"
+    shift
+
+    # remove leading tag prefix
+    container_name="${1#*/}"
+
+    # remove container name suffix
+    echo "${1}" | grep -q "/" && : tag_prefix="${1%/*}"
+
+    # assign default value in case argument did not contain a tag prefix
+    : ${tag_prefix:=nmstate}
+
+    docker build --no-cache ${extra_args} -t "${tag_prefix}/${container_name}" \
```

Comment:
What happened to `--rm` ?

I would also extract this part as well (into a function) and use a variable/constant to add the default arguments in order to make it clear and less hidden.

Reply:
It is not used here: https://github.com/nmstate/nmstate/blob/master/automation/run-tests.sh#L160 and the manpage says it is true by default. I can also add it, I do not care atm.

You want a constant for `--no-cache` and the `tag_prefix`?

Reply:
Now I understood your remark about the prefix and changed it to `nmstate`.